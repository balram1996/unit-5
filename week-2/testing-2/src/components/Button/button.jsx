import React from 'react'

export const Button = ({label,onClick}) => {
    return (
        <>
          <button data-testid="button" onClick={onClick}>{label}</button>  
        </>
    )
}
