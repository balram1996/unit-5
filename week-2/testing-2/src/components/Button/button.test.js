import { Button } from "./button";
import  ReactDOM from "react-dom";
import {render,fireEvent,cleanup} from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup)
it("should be render correctly with hello",()=>{
//     const screen = render(<Button/>);
//    expect(screen).toBeInTheDocument();
const {getByTestId}= render(<Button label="hello"/>);
const button = getByTestId("button");
expect(button).toHaveTextContent("hello")
});



it("show match snapshot",()=>{
    const fn = jest.fn();
    const tree = renderer
    .create(<Button label="hello x" onClick={fn}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
    })