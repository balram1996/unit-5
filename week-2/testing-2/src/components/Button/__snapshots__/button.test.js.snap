// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show match snapshot 1`] = `
<button
  data-testid="button"
  onClick={[MockFunction]}
>
  hello x
</button>
`;
